### ISSUE TITLE
App Crashes "No Photos or Media found on Device"

### ISSUE THREAD

**Issue by ITCMD** [scubajeff/lespas#132]
**Describe the bug**
Sometimes it happens when I open the app, immediately, sometimes it happens when I click gallery, sometimes when I hit the camera folder.

The app instantly closes out and a little popup says "No photos or media found on device"

When this happens, sometimes after trying like 50 times itll start working again. Often it gets further but still crashes. I feel like deleting photos from within another app triggers it. Soemtimes nkthing triggers it. Often I have to restart the phone to get it working.

I can still access the menu and trigger a sync.

**To Reproduce**
1. Open the app
2. See error

**Expected behavior**
I mean, it doesnt crash.

**Screenshots**

![Screenshot_20230628_230301_Device care](https://github.com/scubajeff/lespas/assets/32961763/f00aba88-efc9-4464-b136-f296a83a3881)
![Screenshot_20230628_232247_Nova7](https://github.com/scubajeff/lespas/assets/32961763/0bb2ce0f-6953-49f9-ac78-7e4242a6bda2)

**Smartphone (please complete the following information):**
 - Device: Galaxy S23 Ultra
 - OS: Android 13
 - Version: v2.9.0

**Comment by ITCMD** [comment]
Additional notes: clearing cache does not fix. Logging out and in does not fix.

**Comment by scubajeff** [comment]
I think those crash reports on Google Play Console recently came from you. I have read the crash logs, have no idea yet what went wrong, since crashes happen in Android Framework and the cause is "No Such Method", kind of weird.

**Comment by ITCMD** [comment]
That's very strange. Let me know if you'd like me to try anything. I'm fairly familiar with the Android system and installing APKs and stuff

**Comment by scubajeff** [comment]
Are you using a stock Samsung ROM, or you twists it a bit? All the crash report I got are related to the new Gallery function, did you launch Les Pas from it's Gallery launcher shortcut? Do you encounter crash when you are using features other than the Gallery, like creating albums etc.?

**Comment by scubajeff** [comment]
A new load of crash logs coming in from Google Play console, seems all related to missing the common storage volume 'external_primary'.

**Comment by ITCMD** [comment]
Interesting. On my S23 Ultra theres no support for external storage.

This is now always happening. Sometimes it will loa into the app for a moment, but no pictures load, it's all black icons. I tried to take a screenshot in the app but it said "Cannot take screenshot of content"

**Comment by ITCMD** [comment]
The Samsung rom is stock, as I received it. Im not sure about shortcuts. I open Les Pas, click Gallery, click Camera, or any other foler, and in any subfoler from Gallery when I open the folder all the icons go black (including the gallery subfolders which showed previews of the photos in the folder before clicking there, and it crashes after a moment.

**Comment by scubajeff** [comment]
Please try revoking the Image and Video access permission and grant it again. See if this would help.

**Comment by ITCMD** [comment]
I did that. Opened the app, it prompted for access and promptly crashed again after I accepted.

**Comment by scubajeff** [comment]
So far, all logs show that crashes happened deep down Android's framework code base, and all related to MediaStore access being denied. Various test machine here in my lab running Android 13 don't have this behavior, I don't have Samsung phone though. Suspect that there are some very specific system settings which trigger these crashes. Settings like work profile restriction, Samsung KNOX settings etc.

**Comment by scubajeff** [comment]
Another crash log just pop up on my Google Play console. Seems like this bug is affecting release 2.9.1 on the same device too.

**Comment by ITCMD** [comment]
ah so that is samsung knox

**Comment by scubajeff** [comment]
This is something new for me. Need to do some research on Knox API if it exists.

**Comment by ITCMD** [comment]
Keep me posted. If you want me to test anything please let me know!

Perhaps in the meantime it'd be worth posting an app that is just the sync part. Not sure how easy that would be to do. Your sync works so much better than the official nextcloud one and it still works as long as I dont open the gallery

**Comment by scubajeff** [comment]
> it still works as long as I dont open the gallery

Do you mean the background sync can read your secret folder while the Gallery module can't? If that's the case, it will be very interesting.

**Comment by ITCMD** [comment]
No, it cannot read the secret folder, and there's no indication it's there when browsing it. It's not a browsable folder as far as I know, and even if it was it's all encrypted.
I believe it's a virtual machine basically, because you can add apps to it and it runs in a separate instance and the two have no access to each other unless you move a file between them.

**Comment by ITCMD** [comment]
Curious if there are any updates on this issue. I'm sure you're busy, I'd just love to use this app!

It did seem to work well for the first day or so too.

**Comment by scubajeff** [comment]
I'm having a vacation with my kids right now. There are won't be any updates until early next month.

**Comment by scubajeff** [comment]
I flicked through Samsung's API document on Knox secure folder, as far as I can tell, it's **NOT** feasible. Sorry.

**Comment by ITCMD** [comment]
Interesting. It's crazy that Knox breaks the app so much being in a totally separate OS image. Hopefull Samsung fixes that or better documents it at some point.

Sorry to have bothered you on your vacation. Enjoy your time with your kids!

**Comment by scubajeff** [comment]
This bug might be fixed in release 2.9.2. Please test and report.
